src/test/util_tests.cpp

```diff
@@ -49,24 +49,28 @@ BOOST_FIXTURE_TEST_SUITE(util_tests, BasicTestingSetup)
 
 BOOST_AUTO_TEST_CASE(util_datadir)
 {
-    ClearDatadirCache();
-    const fs::path dd_norm = GetDataDir();
+    // Use local args variable instead of m_args to avoid making assumptions about test setup
+    ArgsManager args;
+    args.ForceSetArg("-datadir", m_path_root.string());
 
-    gArgs.ForceSetArg("-datadir", dd_norm.string() + "/");
-    ClearDatadirCache();
-    BOOST_CHECK_EQUAL(dd_norm, GetDataDir());
+    fs::path dd_norm = args.GetDataDirPath();
 
-    gArgs.ForceSetArg("-datadir", dd_norm.string() + "/.");
-    ClearDatadirCache();
-    BOOST_CHECK_EQUAL(dd_norm, GetDataDir());
-
-    gArgs.ForceSetArg("-datadir", dd_norm.string() + "/./");
-    ClearDatadirCache();
-    BOOST_CHECK_EQUAL(dd_norm, GetDataDir());
-
-    gArgs.ForceSetArg("-datadir", dd_norm.string() + "/.//");
-    ClearDatadirCache();
```

Comment:
In commit "Modify "util_datadir" unit test to not use gArgs." (8584666393255a7ce62d741ddd8d994c486423c6)

It seems like the checks in this test might be meaningless now that the ClearDatadirCache calls are dropped, because the repeated `BOOST_CHECK_EQUAL(dd_norm, args.GetDataDirPath());` checks will keep passing no matter what value `-datadir` is force set to.

I think it would be best for this commit to either keep the Clear calls so the test is changed as little as possible. Or the test could be rewritten to use different expected values or different args objects.

Reply:
You are right. I have put those commands back.